Assets/Scripts/Controller/Pickup Pup/PPGameController.cs

```diff
@@ -181,24 +209,23 @@ public bool TryBuyItem(ShopItem item)
     void buyItem(int value, CurrencyType valueCurrencyType,
         int cost, CurrencyType costCurrencyType)
     {
-        dataController.ChangeCurrencyByType(value, valueCurrencyType);
-        dataController.ChangeCurrencyByType(-cost, costCurrencyType);
+        ConvertCurrency(value, valueCurrencyType, cost, costCurrencyType);
     }
 
     public bool TryAdoptDog(DogDescriptor dog)
     {
-        if(Coins.Amount < dog.CostToAdopt || VacantHomeSlots.Amount <= 0)
+        if(CanAfford(CurrencyType.Coins, dog.CostToAdopt) && CanAfford(CurrencyType.HomeSlots, 1))
         {
-            return false;
-        }
-        AdoptDog(dog);
-        return true;
+            AdoptDog(dog);
+            return true;
+        }        
+        return false;       
     }
 
     void AdoptDog(DogDescriptor dog)
```

Comment:
Why is this not using the TryBuyItem method? I realize dogs aren't a currency. But I feel like it's confusing that there's a different procedure for purchasing them versus anything else in game